File: leetcode/0067-add-binary/solution.py
"""
* Author: 1chooo<hugo970217@gmail.com>
* Problem: 
* Runtime: 0ms (100.00%)
"""

class Solution:
    def addBinary(self, a: str, b: str) -> str:
        ans: str = ""
        i: int = len(a) - 1
        j: int = len(b) - 1
        carry: int = 0

        while (i >= 0 or j >= 0 or carry):
            if i >= 0:
                carry += int(a[i])
                i -= 1
            if j >= 0:
                carry += int(b[j])
                j -= 1

            ans += str(carry % 2)
            carry //= 2

        return ans[::-1]

File: leetcode/0067-add-binary/solution.cpp
/**
 * Author: 1chooo<hugo970217@gmail.com>
 * Problem: https://leetcode.com/problems/add-binary
 * Runtime: 0ms (100.00%)
 */

const static auto _ = []() {
    cin.tie(nullptr)->sync_with_stdio(false);
    return nullptr;
}();

class Solution {
  public:
    string addBinary(string a, string b) {
        string ans = "";
        int i = a.size() - 1;
        int j = b.size() - 1;
        int carry = 0;

        while (i >= 0 || j >= 0 || carry) {
            int sum = carry;
            if (i >= 0)
                sum += a[i--] - '0';
            if (j >= 0)
                sum += b[j--] - '0';

            ans += (sum % 2) + '0';
            carry = sum / 2;
        }

        reverse(ans.begin(), ans.end());

        return ans;
    }
};
